libafl/src/executors/timeout.rs

```diff
@@ -322,11 +296,32 @@ where
     }
 }
 
+#[cfg(unix)]
+impl<E> HasOnCrashReset for TimeoutExecutor<E> {
+    /// Disarm currently running timer
+    fn reset(&self) {
+        unsafe {
+            let disarmed: libc::itimerspec = zeroed();
+            libc::timer_settime(self.timerid, 0, &disarmed as *const _, null_mut());
+        }
+    }
+}
+
+#[cfg(all(windows, feature = "std"))]
+impl<E> HasOnCrashReset for TimeoutExecutor<E> {
+    /// Deletes this timer queue
+    /// # Safety
+    /// Will dereference the given `tp_timer` pointer, unchecked.
+    fn reset(&self) {
+        CloseThreadpoolTimer(self.tp_timer);
+    }
+}
+
 #[cfg(windows)]
 impl<E> Drop for TimeoutExecutor<E> {
     fn drop(&mut self) {
         unsafe {
-            windows_delete_timer_queue(self.tp_timer);
+            self.reset();
```

Comment:
I'd leave the original delete here, in case we call through to some other `reset` fn on crash

Reply:
could you elaborate on this?
On second thought, I am not sure anymore what exactly you mean.

Reply:
We should call through to the inner `reset` in the normal reset fn. But this is the destructor so we shouldn't call the inner resets here

Reply:
I am not sure why I added this Drop in the first place... 😓 
Now I think we don't even need this destructor